### ISSUE TITLE
qmstr.yaml configuration instructions

### ISSUE THREAD

**Issue by utkarshyen** [QMSTR/qmstr#469]
Hello @mirkoboehm 

I have followed the installation steps mentioned in document.
Tested the tutorials, debian curl and json-c packages.

However, I could not see the binary licenses (target licenses) in scancode.json file, even in report I could see all the tabs as BLANK.

Please find the qmstr.yaml file.
**NOTE** I have used same qmstr.yaml file from json-c tutorials for testing purpose.
----------------------------------------------------------------------------------------
project:
  name: "elfutils"
  metadata:
    Vendor: ""
    OcFossLiaison: ""
    OcComplianceContact: ""
  analysis:
    - analyzer: spdx-identifier-analyzer
      name: "Simple SPDX Analyzer"
      config:
        workdir: "/buildroot"
    - analyzer: scancode-analyzer
      name: "Scancode Analyzer"
      config:
        workdir: "/buildroot"
        resultfile: "/buildroot/scancode.json"
        #cached: "true"
    - analyzer: test-analyzer
      name: "Simple CI Test Analyzer"
      config:
        workdir: "/buildroot"
  reporting:
    - reporter: test-reporter
      name: "Test Reporter"
      config:
        siteprovider: "Endocode"
    - reporter: package-manifest-reporter
      name: "Package manifest Reporter"
      config:
        outputdir: "/buildroot"
        pathSubst: ".+?/.+?/.+?/(.+)||./$1"
    - reporter: qmstr-reporter-html
      name: "Public HTML Reports"
      config:
        siteprovider: "Endocode"
        baseurl: "http://qmstr.org/packages/"
----------------------------------------------------------------------------------------

Can you please put some light on this? Am I missing any steps here? Or any configuration setting?
Also, https://qmstr.org/documentation/introduction/getting-started/Qmstr.yaml.md page is not found.

Regards,
Utkarsh

**Comment by utkarshyen** [comment]
@senaingrid Can you please provide your views on this?

**Comment by utkarshyen** [comment]
Any update on this?

**Comment by senaingrid** [comment]
Hi @utkarshyen, which document are you following? It seems you're using an outdated version of the config file. You can find the yaml example file at this link: https://qmstr.org/documentation/configuration/

Best,
Ingrid

**Comment by utkarshyen** [comment]
Hello @senaingrid 

I followed the same document which you mentioned for config file
> https://qmstr.org/documentation/configuration/

I'm trying to look the issue at my end. I could not see any entries in generated HTML report.
Also, some of the analyzers are failing for some of the packages.

Regards,
UT

**Comment by senaingrid** [comment]
Are you using a public fork of `elfutils`? Can you share the link so I can try to replicate the scenario here?

**Comment by utkarshyen** [comment]
I'm using ubuntu 18.04 debian source,
"elfutils-0.170", building it with make instead of debian build (dpkg-buildpackage) framework.

Regards,

**Comment by GiasemiSh** [comment]
Hi @utkarshyen. Sorry in advance, our documentation is not complete but we are trying our best.
--
You tried the tutorial, debian curl and json-c packages. Did they work for you? 
You mention that you could not see the binary licenses. Was the scancode.json empty?

**Comment by utkarshyen** [comment]
Hello @GiasemiSh,

Sorry to bother.
Yes. I tried both the tutorials debian curl and json-c packages. However,
**debian curl package** - analyze fails for spdx-identifier-analyzer as well as for scancode-analyzer.
Thus no report.
**json-c package** - Everything work fine. But the report generated in blank. No license for binary.
I checked in scancode.json there is no entry for binary. Rest contents are OKAY!

On the other note, I have explored the source code of QuaterMaster. I came to below understanding,
QuaterMaster invokes ScanCode for licensing and ScanCode invokes strings to get the license from binary. Please note, I have also explored the Scancode.
So basically, what output ''$strings -a <binary_name> | grep license'' gives is the license for the binary.
Please correct if my understanding is wrong.
Do QuaterMaster gives the information from which source files this license has come into the binary?

Regards,

**Comment by GiasemiSh** [comment]
Scancode detects licenses from the source files. 
Quartermaster creates a build graph, from the source files to the resulting package. 
We add the license information we have collected from ScanCode to the source files and through the build graph we know which licenses come into the binary or the package as a whole.

Now if you want to view the results I would ask you to make a change in your yaml configuration.

--> *In order to deploy this option you need to have [Hugo](https://gohugo.io/getting-started/installing/) installed in your machine.*

In the qmstr-reporter-html configuration please add the following line:
` generatehtml: "no"`

So the qmstr-reporter-html configuration would look like this:
```
- reporter: qmstr-reporter-html
      name: "HTML Reporter"
      config:
        generatehtml: "no"
        siteprovider: "Endocode"
        baseurl: "http://localhost:8080/"
```

With this option, you disable qmstr to execute Hugo which generates the static HTML page with the QMSTR reports. Since it's disabled you can afterwards run it locally and view the results.


Please build the **json-c package** with QMSTR again.
After QMSTR has finished and quit, you can go to the qmstr-reporter-html output, execute Hugo locally and view the results:

```
> cd qmstr/qmstr-reporter-html/HTML_Reporter
> ls
qmstr-reports.tar.bz2

> tar -xf qmstr-reports.tar.bz2
> cd qmstr-your_random_id
> hugo server

You should be able to view the report results at http://localhost:1313/

**Comment by utkarshyen** [comment]
Hello @GiasemiSh ,

Thank you for your detailed info. It was really helpful.
Also tried your config changes.
However, I could see the page on local host but data was not present. Even from hugo server I could not see any entries in the report.

I checked the scancode.json and it is fine.
PFA ss of the report.
Meanwhile investigating on my side.

Regards,

![elfutils_report](https://user-images.githubusercontent.com/59991469/77043917-7daa6380-69e4-11ea-9efe-f90dc11fbd4f.png)

**Comment by utkarshyen** [comment]
Hello,

>We add the license information we have collected from ScanCode to the source files and through the build graph we know which licenses come into the binary or the package as a whole.

I have seen some sample report of curl on https://qmstr.org/packages/curl/curl/curl-7_64_1-142-g761b49b91/ which gives the licenses for binary. But no info of source list used or which license belongs to which source files. 
Am I missing anything here?

For e.g.
filea licensea
fileb licenseb
...
...
filen licensen

However, binary is built with only filea and thus licensea. Does QuaterMaster provides such information. I could answer these questions myself once I successfully setup and run the QuaterMaster my side and explore. Just sharing my view.

Regards,
